stacks-signer/src/signer.rs

```diff
@@ -0,0 +1,2005 @@
+                    .stackerdb
+                    .send_message_with_retry(block_rejection.into())
+                {
+                    warn!(
+                        "Signer #{}: Failed to send block submission to stacker-db: {e:?}",
+                        self.signer_id
+                    );
+                }
+            }
+            SignError::Aggregator(e) => {
+                warn!(
+                    "Signer #{}: Received an aggregator error: {e:?}",
+                    self.signer_id
+                );
+            }
+        }
+        // TODO: should reattempt to sign the block here or should we just broadcast a rejection or do nothing and wait for the signers to propose a new block?
+    }
+
+    /// Send any operation results across the provided channel
+    fn send_operation_results(
+        &mut self,
+        res: Sender<Vec<OperationResult>>,
+        operation_results: Vec<OperationResult>,
+    ) {
+        let nmb_results = operation_results.len();
+        match res.send(operation_results) {
+            Ok(_) => {
+                debug!(
+                    "Signer #{}: Successfully sent {} operation result(s)",
+                    self.signer_id, nmb_results
+                )
+            }
+            Err(e) => {
+                warn!(
+                    "Signer #{}: Failed to send {nmb_results} operation results: {e:?}",
+                    self.signer_id
+                );
+            }
+        }
+    }
+
+    /// Sending all provided packets through stackerdb with a retry
+    fn send_outbound_messages(&mut self, outbound_messages: Vec<Packet>) {
+        debug!(
+            "Signer #{}: Sending {} messages to other stacker-db instances.",
+            self.signer_id,
+            outbound_messages.len()
+        );
+        for msg in outbound_messages {
+            let ack = self.stackerdb.send_message_with_retry(msg.into());
+            if let Ok(ack) = ack {
+                debug!("Signer #{}: send outbound ACK: {ack:?}", self.signer_id);
+            } else {
+                warn!(
+                    "Signer #{}: Failed to send message to stacker-db instance: {ack:?}",
+                    self.signer_id
+                );
+            }
+        }
+    }
+
+    /// Update the DKG for the provided signer info, triggering it if required
+    pub fn update_dkg(
+        &mut self,
+        stacks_client: &StacksClient,
+        current_reward_cycle: u64,
+    ) -> Result<(), ClientError> {
+        let reward_cycle = self.reward_cycle;
+        let new_aggregate_public_key = stacks_client.get_approved_aggregate_key(reward_cycle)?;
+        let old_aggregate_public_key = self.coordinator.get_aggregate_public_key();
+        if new_aggregate_public_key.is_some()
+            && old_aggregate_public_key != new_aggregate_public_key
+        {
+            // TODO: this will never work as is. We need to have stored our party shares on the side etc for this particular aggregate key.
```

Comment:
Is there a ticket for this?

Reply:
Yes there is. https://github.com/stacks-network/stacks-core/issues/4388

It is the Ticket hank has just started looking into. Requires changes on wsts side.